src/testResultsFile.ts

```diff
@@ -0,0 +1,74 @@
+"use strict";
+import * as fs from "fs";
+import * as path from "path";
+import { setTimeout } from "timers";
+import { Disposable, Event, EventEmitter } from "vscode";
+import { DOMParser } from "xmldom";
+import { TestResult } from "./testResult";
+
+function getAttributeValue(node, name: string): string {
+    const attribute = node.attributes.getNamedItem(name);
+    return (attribute === null) ? null : attribute.nodeValue;
+}
+
+export class TestResultsFile implements Disposable {
+    private static readonly ResultsFileName = "TestExplorerResults.trx";
```

Comment:
What if we have multiple test project opened in different VS Code?

Reply:
I'm now using a temporary random folder to store the results, so now works across multiple instances